BuildMacOSUniveralBinary.py

```diff
@@ -0,0 +1,146 @@
+  os.mkdir("x64");
+  os.chdir("x64");
+
+  os.system('PKG_CONFIG_PATH="'+x64_pkg_config_path+'"  Qt5_DIR="'+x64_qt5_path+'" CMAKE_OSX_ARCHITECTURES=x86_64 arch -x86_64 cmake ../../ -G Xcode')
+  os.chdir("..");       
+
+#Build ARM and x64 projects 
+os.system('xcodebuild -project arm/dolphin-emu.xcodeproj -target "'+build_target+'" -configuration Release');
+os.system('xcodebuild -project x64/dolphin-emu.xcodeproj -target "'+build_target+'" -configuration Release');
+
+#Merge ARM and x64 binaries into universal binaries
+
+#Source binaries to merge together
+src_app0 = "arm/Binaries/release"
+src_app1 = "x64/Binaries/release"
+
+
+if os.path.exists(dst_app): shutil.rmtree(dst_app)
+os.mkdir(dst_app);
+
+def lipo(path0,path1,dst):
+  cmd = 'lipo -create -output "'+dst + '" "' + path0 +'" "' + path1+'"'
+  print(cmd)
+  os.system(cmd)
+def recursiveMergeBinaries(src0,src1,dst):
+  #loop over all files in src0
+  for newpath0 in glob.glob(src0+"/*"):
+    filename = os.path.basename(newpath0);
+    newpath1 = os.path.join(src1,filename);
+    new_dst_path = os.path.join(dst,filename);
+    if not os.path.islink(newpath0):
+      if os.path.exists(newpath1):
+        if os.path.isdir(newpath1):
+          os.mkdir(new_dst_path);
+          #recurse into directories
+          recursiveMergeBinaries(newpath0,newpath1,new_dst_path)
+        else:
+          if filecmp.cmp(newpath0,newpath1):
+            #copy files that are the same
+            shutil.copy(newpath0,new_dst_path);
+          else:
+            #lipo together files that are different
+            lipo(newpath0,newpath1,new_dst_path)
+      else:
+        #copy files that don't exist in path1
+        shutil.copy(newpath0,new_dst_path)
+  
+  #loop over files in src1 and copy missing things over to dst
+  for newpath1 in glob.glob(src1+"/*"):
+    filename = os.path.basename(newpath0);
+    newpath0 = os.path.join(src0,filename);
+    new_dst_path = os.path.join(dst,filename);
+    if not os.path.exists(newpath0) and not os.path.islink(newpath1):
+      shutil.copytree(newpath1,new_dst_path);  
+  
+  #fix up symlinks for path0
+  for newpath0 in glob.glob(src0+"/*"):
+    filename = os.path.basename(newpath0);
+    new_dst_path = os.path.join(dst,filename);
+    if os.path.islink(newpath0):
+      relative_path = os.path.relpath(os.path.realpath(newpath0),src0)
+      print(relative_path,new_dst_path)
+      os.symlink(relative_path,new_dst_path);
+  #fix up symlinks for path1
+  for newpath1 in glob.glob(src1+"/*"):
+    filename = os.path.basename(newpath1);
+    new_dst_path = os.path.join(dst,filename);
+    newpath0 = os.path.join(src0,filename);
+    if os.path.islink(newpath1) and not os.path.exists(newpath0):
+      relative_path = os.path.relpath(os.path.realpath(newpath1),src1)
+      print(relative_path,new_dst_path)
+      os.symlink(relative_path,new_dst_path);
+                                            
+  return;
+
+#create univeral binary
```

Comment:
Ideally, the script should have a main function and shouldn't be doing anything if not executed (i.e. shouldn't be building if loaded via an `import` statement). That's useful for several reasons, like debugging, tinkering  or loading specific functions from a module.

Example:
```python
if __name__ == "__main__":
    main()
```